File: Programmers/해시/전화번호 목록/phone_num_list.py
def solution(phone_book):
    answer=True
    hash_map={}
    for phone_num in phone_book:
        hash_map[phone_num]=1
    for phone_num in phone_book:
        tmp=""
        for num in phone_num:
            tmp+=num
            if tmp in hash_map and tmp!=phone_num:
                answer=False
    return answer



print(solution(["12","123","1235","567","88"]))

File: Programmers/해시/전화번호 목록/phone_num_list.js
function solution(phone_book){
    let answer=true
    dic={}

    for(let p of phone_book){
        dic[p]=1
    }
    for(let p of phone_book){
        let tmp='';
        for(let num of p){
            tmp+=num;
            if(tmp in dic && tmp!==p){
                answer=false
            }
        }
    }
    return answer
}

const phone_book=["12","123","1235","567","88"]	;
console.log(solution(phone_book));